build.gradle

```diff
@@ -57,50 +63,45 @@ nexusPublishing {
             password.set(System.getenv("MAVEN_PASSWORD"))
         }
     }
+    transitionCheckOptions {
+        maxRetries.set(60)
+        delayBetween.set(Duration.ofSeconds(30))
+    }
 }
 
 publishing {
     publications {
-        mavenJava(MavenPublication) {
-            artifactId = 'license-gradle-plugin'
-            from components.java
-
+        withType(MavenPublication) {
             pom {
-                name = 'Octopus License Gradle Plugin'
-                description = 'A plugin for license management in Gradle projects.'
-                url = 'https://github.com/octopusden/octopus-license-gradle-plugin.git'
+                name = project.name
+                description = "Octopus module for license gradle plugin"
+                url = "https://github.com/octopusden/octopus-license-gradle-plugin.git"
                 licenses {
                     license {
-                        name = 'The Apache License, Version 2.0'
-                        url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                        name = "The Apache License, Version 2.0"
+                        url = "http://www.apache.org/licenses/LICENSE-2.0.txt"
                     }
                 }
+                scm {
+                    url = "https://github.com/octopusden/octopus-license-gradle-plugin.git"
+                    connection = "scm:git://github.com/octopusden/octopus-license-gradle-plugin.git"
+                }
                 developers {
                     developer {
-                        id = 'octopus'
-                        name = 'Octopus Team'
-                        email = 'team@octopusden.org'
+                        id = "octopus"
+                        name = "octopus"
                     }
                 }
-                scm {
-                    connection = 'scm:git:git://github.com/octopusden/octopus-license-gradle-plugin.git'
-                    developerConnection = 'scm:git:ssh://github.com/octopusden/octopus-license-gradle-plugin.git'
-                    url = 'https://github.com/octopusden/octopus-license-gradle-plugin'
-                }
             }
         }
     }
 }
 
-signing {
-    def signingKey = project.findProperty('signingKey')
-    def signingPassword = project.findProperty('signingPassword')
-
-    useInMemoryPgpKeys(signingKey, signingPassword)
-
-    gradle.taskGraph.whenReady { taskGraph ->
-        if (taskGraph.hasTask(':publishToSonatype')) {
-            sign publishing.publications.mavenJava
-        }
+if (project.hasProperty("nexus") && !project.version.toString().toUpperCase().endsWith("SNAPSHOT")) {
```

Comment:
@adellinekania I consider that as a bad solutions.

Why not https://github.com/octopusden/octopus-teamcity-automation/blob/08571ae1282e2605315eb580413b1afc2718d486/build.gradle.kts#L156 ?

Reply:
Okay, will do